cores/cosa/Cosa/INET/SNMP.cpp

```diff
@@ -114,11 +115,14 @@ int
 SNMP::OID::match(const uint8_t* coid, bool flag)
 {
   uint8_t clen = pgm_read_byte(&coid[0]);
-  if (length < clen) return (-1);
-  for (uint8_t i = 0; i < clen; i++) 
-    if (name[i] != pgm_read_byte(&coid[i + 1])) return (-1);
+  for (uint8_t i = 0; i < clen; i++) {
+    if (i >= length) return (-1);
+    uint8_t coidb = pgm_read_byte(&coid[i + 1]);
+    if (name[i] < coidb) return (-1);
+    if (name[i] > coidb) return (-2);
```

Comment:
I agree, a total hack and def needs to be documented if we go this way but here I have the match() method return -2 if the MIB comes later in MIB sequence than coid; -1 means before, 0 means is the same as and +ve values are either the the last byte of the MIB or length depending on the flag.

Not my proudest moment of coding but I was trying to maintain the Cosa SNMP API whilst being able get some idea of before and after.

Sorry for not updating docs but have not sussed out the way they are generated yet - I should have at least explained in a comment but at that point I was just happy it was working as it should. Maybe a whole new method is a better way to do this and maintain backwards compatibility but also very interested in your ideas.